tests/production_pause_max_rev_blks_test.py

```diff
@@ -0,0 +1,166 @@
+    # each producer node has finalizer configured.
+    # In this test, node1 does not have finalizer configured.
+    if cluster.launch(pnodes=pnodes, totalNodes=totalNodes,
+                      totalProducers=pnodes, prodCount=prodCount,
+                      delay=delay, loadSystemContract=False,
+                      extraNodeosArgs=extraNodeosArgs,
+                      activateIF=False, signatureProviderForNonProducer=True,
+                      topo="./tests/production_pause_max_rev_blks_test_shape.json") is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    assert cluster.biosNode.getInfo(exitOnError=True)["head_block_producer"] != "eosio", "launch should have waited for production to change"
+
+    node0          = cluster.getNode(0)   # producer and finalizer node for defproducera
+    producerbNode  = cluster.getNode(1)   # producer node for defproducerb
+    finalizerbNode = cluster.getNode(2)   # finalizer node for defproducerb
+
+    Print("Set finalizer policy and start transition to Savanna")
+    # Specifically, need to configure finalizer name for finalizerbNode as defproducerb
+    transId = cluster.setFinalizers(nodes=[node0, finalizerbNode], finalizerNames=["defproducera", "defproducerb"])
+    assert transId is not None, "setfinalizers failed"
+    assert cluster.biosNode.waitForTransFinalization(transId), f"setfinalizers transaction {transId} was not rolled into a LIB block"
+    assert cluster.biosNode.waitForLibToAdvance(), "LIB did not advance after setFinalizers"
+
+    # biosNode no longer needed
+    cluster.biosNode.kill(signal.SIGTERM)
+    cluster.waitOnClusterSync(blockAdvancing=5)
+
+    Print("Wait for LIB on all producing nodes to advance")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert producerbNode.waitForLibToAdvance(), "producerbNode did not advance LIB"
+
+    ####################### test 1 ######################
+    # production paused due to max reversible blocks exceeded
+
+    Print("Shutdown finalizerbNode")
+    finalizerbNode.kill(signal.SIGTERM)
+    assert not finalizerbNode.verifyAlive(), "finalizerbNode did not shutdown"
+
+    # Verify LIB stalled on node0 and producerbNode due to finalizerNode was shutdown
+    Print("Verify LIB stalled  after shutdown of finalizerbNode")
+    if producerbNode.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not producerbNode.waitForLibToAdvance(timeout=5), "LIB should not advance on producerbNode after finalizerbNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after finalizerbNode was shutdown"
+
+    # give time for max reversible blocks to be exceeded
+    time.sleep(24) # 24 is the time to produce 48 blocks which is the configured max-reversible-blocks
```

Comment:
Instead of a sleep, wait for a head block above lib that exceeds max reversible.